subsys/net/lib/openthread/platform/flash.c

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <flash.h>
+
+#include "platform-zephyr.h"
+
+static struct device *flash_dev;
+static size_t ot_flash_size;
+static size_t ot_flash_offset;
+
+static inline u32_t mapAddress(u32_t aAddress)
+{
+	return aAddress + ot_flash_offset;
+}
+
+otError utilsFlashInit(void)
+{
+	int i;
+	struct flash_pages_info info;
+	size_t pages_count;
+
+	flash_dev = device_get_binding(
+			CONFIG_OT_PLAT_FLASH_DEVICE_NAME);
+
+	if (!flash_dev) {
+		return OT_ERROR_NOT_IMPLEMENTED;
+	}
+
+	pages_count = flash_get_page_count(flash_dev);
+
+	if (flash_get_page_info_by_idx(flash_dev,
+		pages_count - CONFIG_OT_PLAT_FLASH_PAGES_COUNT - 1, &info)) {
+
+		return OT_ERROR_FAILED;
+	}
+
+	ot_flash_offset = info.start_offset;
+	ot_flash_size = 0;
+
+	for (i = 0; i < CONFIG_OT_PLAT_FLASH_PAGES_COUNT; i++) {
+		if (flash_get_page_info_by_idx(flash_dev,
+			pages_count - i - 1, &info)) {
+
+			return OT_ERROR_FAILED;
+		}
+		ot_flash_size += info.size;
+	}
+
+	return OT_ERROR_NONE;
+}
+
+u32_t utilsFlashGetSize(void)
+{
+	return ot_flash_size;
+}
+
+otError utilsFlashErasePage(u32_t aAddress)
+{
+	struct flash_pages_info info;
+	u32_t address;
+
+	address = mapAddress(aAddress);
+	if (flash_get_page_info_by_offs(flash_dev, address, &info)) {
+		return OT_ERROR_FAILED;
+	}
+
+	if (flash_erase(flash_dev, address, info.size)) {
+		return OT_ERROR_FAILED;
+	}
+
+	return OT_ERROR_NONE;
+}
+
+otError utilsFlashStatusWait(u32_t aTimeout)
+{
+	ARG_UNUSED(aTimeout);
+
+	return OT_ERROR_NONE;
+}
+
+u32_t utilsFlashWrite(u32_t aAddress, uint8_t *aData, u32_t aSize)
```

Comment:
@KaSroka It does not take into account the `write_block_size` and so it will not work on all platforms, e.g. not on NXP K64F. aSize is probably always multiple of 4, is it configurable? I did not find anything suitable in the OT examples.

Reply:
It is at least multiple of 4 as far as I can see in OT code but it's not configurable.